GVFS/GVFS.Common/Maintenance/GitMaintenanceStep.cs

```diff
@@ -132,9 +135,16 @@ protected GitProcess.Result RunGitCommand(Func<GitProcess, GitProcess.Result> wo
 
                 if (!this.Stopping && result?.ExitCodeIsFailure == true)
                 {
+                    string errorMessage = result.Errors;

```

Comment:
Is `result.Errors` guaranteed to be non-null?

Reply:
Ha, no, nice catch.

We creatively call this which will result in null:

```
            this.RunGitCommand(
                process =>
                {
                    this.TryPrefetchCommitsAndTrees(out error, process);
                    return null;
                },
                "TryPrefetchCommitsAndTrees");
```

Reply:
> In that case you'll never get into the if.

I noticed that too.  Unfortunately the GitProcess.Result class itself didn't have any hints about whether Errors would always be non-null and I agree it's better to be safe.